mip/mip.c

```diff
@@ -970,14 +972,11 @@ static struct queue qp;
 // TODO(scaprile): avoid concurrency issues (2 queues ?)
 void qp_mark(unsigned int type, int len) {
   static bool ovf = false;
-  static uint16_t drop_ctr = 0, irq_ctr = 0;
-  struct qpentry e = {
-      .timestamp = mg_millis(), .type = (uint16_t) type, .len = (uint16_t) len};
-
-  if (type == QP_IRQTRIGGERED)
-    e.len = ++irq_ctr;  // only incremented on IRQ calls
-  else if (type == QP_FRAMEDROPPED)
-    ++drop_ctr;  // only incremented on IRQ calls
+  static uint16_t irq_ctr = 0, drop_ctr = 0;
+  struct qpentry e = {.timestamp = (uint32_t) mg_millis(),
+                      .type = (uint16_t) type,
+                      .len = (uint16_t) len};
+  if (type == QP_IRQTRIGGERED) e.len = ++irq_ctr;
   if (ovf) {
     e.type = (uint16_t) QP_QUEUEOVF;
     e.len = drop_ctr;
```

Comment:
drop_ctr is local and never incremented
(and we can't access ifp->dropped from here)